components/script/dom/document.rs

```diff
@@ -1935,6 +1960,61 @@ impl Document {
 
         self.window.layout().nodes_from_point_response()
     }
+
+    // https://html.spec.whatwg.org/multipage/#sequential-focus-navigation
+    fn sequential_focus_navigation(&self, event: &KeyboardEvent) {
+        // Step 1
+        // Step 2
+        // TODO: Implement the sequential focus navigation starting point.
+        // This can be used to change the starting point for navigation.
```

Comment:
@jdm I'm stuck by the `add_focusable_element` and `remove_focusable_element` methods above.

According to the spec,

> When the user requests that focus move from the currently focused area of a top-level browsing context to the next or previous focusable area (e.g. as the default action of pressing the tab key), or when the user requests that focus sequentially move to a top-level browsing context in the first place (e.g. from the browser's location bar), the user agent must use the following algorithm

Thus, maybe we don't need to save a `focus_list` inside `Document`? We might only need to know the _`starting point`_?

I'm wondering if it's possible to calculate the _`starting point`_ and also its next and previous focus candidate.

Reply:
If we store the node for the starting point and not the list and starting index then it might be easier, like you suggest. It would require:
* determining and storing the new starting point if the current starting point ever becomes unfocusable
* recalculating the list every time we need to change the current starting point

Reply:
@jdm Thanks for your confirmation. I'll try to only store the _starting point_ for it!